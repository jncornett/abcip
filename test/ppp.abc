d ( stack="eth:pppoe:ppp:ip4:tcp" )
c ( a=50004 )

a(syn)
b(syn, ack)
a(ack)

a(ack, len=100)
b(ack, len=100)

a(fin,ack)
b(fin,ack)
a(ack)

d ( stack="eth:pppoe:ppp:ip6:tcp" )
c ( a=50006 )

a(syn)
b(syn, ack)
a(ack)

a(ack, len=100)
b(ack, len=100)

a(fin,ack)
b(fin,ack)
a(ack)

