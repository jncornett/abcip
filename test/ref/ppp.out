 [ver 0] [type 0]PPPoE  [ses 0x2] IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.5.50004 > 10.9.8.7.8: Flags [S], cksum 0xae6f (correct), seq 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 8864 0000  .............d..
	0x0010:  0002 002a 0021 4500 0028 0001 0000 4006  ...*.!E..(....@.
	0x0020:  5cba 0a01 0205 0a09 0807 c354 0008 0000  \..........T....
	0x0030:  0001 0000 0000 5002 2000 ae6f 0000       ......P....o..
 [ver 0] [type 0]PPPoE  [ses 0x2] IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.5.50004: Flags [S.], cksum 0xae5d (correct), seq 1, ack 2, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 8864 0000  .............d..
	0x0010:  0002 002a 0021 4500 0028 0001 0000 4006  ...*.!E..(....@.
	0x0020:  5cba 0a09 0807 0a01 0205 0008 c354 0000  \............T..
	0x0030:  0001 0000 0002 5012 2000 ae5d 0000       ......P....]..
 [ver 0] [type 0]PPPoE  [ses 0x2] IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.5.50004 > 10.9.8.7.8: Flags [.], cksum 0xae5e (correct), ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 8864 0000  .............d..
	0x0010:  0002 002a 0021 4500 0028 0002 0000 4006  ...*.!E..(....@.
	0x0020:  5cb9 0a01 0205 0a09 0807 c354 0008 0000  \..........T....
	0x0030:  0002 0000 0002 5010 2000 ae5e 0000       ......P....^..
 [ver 0] [type 0]PPPoE  [ses 0x2] IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 140)
    10.1.2.5.50004 > 10.9.8.7.8: Flags [.], cksum 0x9a78 (correct), seq 1:101, ack 1, win 8192, length 100
	0x0000:  0209 0807 0605 0201 0203 0405 8864 0000  .............d..
	0x0010:  0002 008e 0021 4500 008c 0003 0000 4006  .....!E.......@.
	0x0020:  5c54 0a01 0205 0a09 0807 c354 0008 0000  \T.........T....
	0x0030:  0002 0000 0002 5010 2000 9a78 0000 4c6f  ......P....x..Lo
	0x0040:  7265 6d20 6970 7375 6d20 646f 6c6f 7220  rem.ipsum.dolor.
	0x0050:  7369 7420 616d 6574 2c20 636f 6e73 6563  sit.amet,.consec
	0x0060:  7465 7475 7220 6164 6970 6973 6369 6e67  tetur.adipiscing
	0x0070:  2065 6c69 742e 2045 7469 616d 2073 7573  .elit..Etiam.sus
	0x0080:  6369 7069 742c 2071 7561 6d20 6174 2073  cipit,.quam.at.s
	0x0090:  6365 6c65 7269 7371 7565 2075 6c74 7269  celerisque.ultri
	0x00a0:  6369                                     ci
 [ver 0] [type 0]PPPoE  [ses 0x2] IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 140)
    10.9.8.7.8 > 10.1.2.5.50004: Flags [.], cksum 0x9a14 (correct), seq 1:101, ack 101, win 8192, length 100
	0x0000:  0201 0203 0405 0209 0807 0605 8864 0000  .............d..
	0x0010:  0002 008e 0021 4500 008c 0002 0000 4006  .....!E.......@.
	0x0020:  5c55 0a09 0807 0a01 0205 0008 c354 0000  \U...........T..
	0x0030:  0002 0000 0066 5010 2000 9a14 0000 4c6f  .....fP.......Lo
	0x0040:  7265 6d20 6970 7375 6d20 646f 6c6f 7220  rem.ipsum.dolor.
	0x0050:  7369 7420 616d 6574 2c20 636f 6e73 6563  sit.amet,.consec
	0x0060:  7465 7475 7220 6164 6970 6973 6369 6e67  tetur.adipiscing
	0x0070:  2065 6c69 742e 2045 7469 616d 2073 7573  .elit..Etiam.sus
	0x0080:  6369 7069 742c 2071 7561 6d20 6174 2073  cipit,.quam.at.s
	0x0090:  6365 6c65 7269 7371 7565 2075 6c74 7269  celerisque.ultri
	0x00a0:  6369                                     ci
 [ver 0] [type 0]PPPoE  [ses 0x2] IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.5.50004 > 10.9.8.7.8: Flags [F.], cksum 0xad95 (correct), seq 101, ack 101, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 8864 0000  .............d..
	0x0010:  0002 002a 0021 4500 0028 0004 0000 4006  ...*.!E..(....@.
	0x0020:  5cb7 0a01 0205 0a09 0807 c354 0008 0000  \..........T....
	0x0030:  0066 0000 0066 5011 2000 ad95 0000       .f...fP.......
 [ver 0] [type 0]PPPoE  [ses 0x2] IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.5.50004: Flags [F.], cksum 0xad94 (correct), seq 101, ack 102, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 8864 0000  .............d..
	0x0010:  0002 002a 0021 4500 0028 0003 0000 4006  ...*.!E..(....@.
	0x0020:  5cb8 0a09 0807 0a01 0205 0008 c354 0000  \............T..
	0x0030:  0066 0000 0067 5011 2000 ad94 0000       .f...gP.......
 [ver 0] [type 0]PPPoE  [ses 0x2] IP (tos 0x0, ttl 64, id 5, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.5.50004 > 10.9.8.7.8: Flags [.], cksum 0xad94 (correct), ack 102, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 8864 0000  .............d..
	0x0010:  0002 002a 0021 4500 0028 0005 0000 4006  ...*.!E..(....@.
	0x0020:  5cb6 0a01 0205 0a09 0807 c354 0008 0000  \..........T....
	0x0030:  0067 0000 0067 5010 2000 ad94 0000       .g...gP.......
 [ver 0] [type 0]PPPoE  [ses 0x2] IP6 (flowlabel 0x00001, hlim 64, next-header TCP (6) payload length: 20) ::ffff:10.1.2.5.50006 > ::ffff:10.9.8.7.8: Flags [S], cksum 0xae6d (correct), seq 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 8864 0000  .............d..
	0x0010:  0002 003e 0057 6000 0001 0014 0640 0000  ...>.W`......@..
	0x0020:  0000 0000 0000 0000 ffff 0a01 0205 0000  ................
	0x0030:  0000 0000 0000 0000 ffff 0a09 0807 c356  ...............V
	0x0040:  0008 0000 0001 0000 0000 5002 2000 ae6d  ..........P....m
	0x0050:  0000                                     ..
 [ver 0] [type 0]PPPoE  [ses 0x2] IP6 (flowlabel 0x00001, hlim 64, next-header TCP (6) payload length: 20) ::ffff:10.9.8.7.8 > ::ffff:10.1.2.5.50006: Flags [S.], cksum 0xae5b (correct), seq 1, ack 2, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 8864 0000  .............d..
	0x0010:  0002 003e 0057 6000 0001 0014 0640 0000  ...>.W`......@..
	0x0020:  0000 0000 0000 0000 ffff 0a09 0807 0000  ................
	0x0030:  0000 0000 0000 0000 ffff 0a01 0205 0008  ................
	0x0040:  c356 0000 0001 0000 0002 5012 2000 ae5b  .V........P....[
	0x0050:  0000                                     ..
 [ver 0] [type 0]PPPoE  [ses 0x2] IP6 (flowlabel 0x00001, hlim 64, next-header TCP (6) payload length: 20) ::ffff:10.1.2.5.50006 > ::ffff:10.9.8.7.8: Flags [.], cksum 0xae5c (correct), ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 8864 0000  .............d..
	0x0010:  0002 003e 0057 6000 0001 0014 0640 0000  ...>.W`......@..
	0x0020:  0000 0000 0000 0000 ffff 0a01 0205 0000  ................
	0x0030:  0000 0000 0000 0000 ffff 0a09 0807 c356  ...............V
	0x0040:  0008 0000 0002 0000 0002 5010 2000 ae5c  ..........P....\
	0x0050:  0000                                     ..
 [ver 0] [type 0]PPPoE  [ses 0x2] IP6 (flowlabel 0x00001, hlim 64, next-header TCP (6) payload length: 120) ::ffff:10.1.2.5.50006 > ::ffff:10.9.8.7.8: Flags [.], cksum 0x9a76 (correct), seq 1:101, ack 1, win 8192, length 100
	0x0000:  0209 0807 0605 0201 0203 0405 8864 0000  .............d..
	0x0010:  0002 00a2 0057 6000 0001 0078 0640 0000  .....W`....x.@..
	0x0020:  0000 0000 0000 0000 ffff 0a01 0205 0000  ................
	0x0030:  0000 0000 0000 0000 ffff 0a09 0807 c356  ...............V
	0x0040:  0008 0000 0002 0000 0002 5010 2000 9a76  ..........P....v
	0x0050:  0000 4c6f 7265 6d20 6970 7375 6d20 646f  ..Lorem.ipsum.do
	0x0060:  6c6f 7220 7369 7420 616d 6574 2c20 636f  lor.sit.amet,.co
	0x0070:  6e73 6563 7465 7475 7220 6164 6970 6973  nsectetur.adipis
	0x0080:  6369 6e67 2065 6c69 742e 2045 7469 616d  cing.elit..Etiam
	0x0090:  2073 7573 6369 7069 742c 2071 7561 6d20  .suscipit,.quam.
	0x00a0:  6174 2073 6365 6c65 7269 7371 7565 2075  at.scelerisque.u
	0x00b0:  6c74 7269 6369                           ltrici
 [ver 0] [type 0]PPPoE  [ses 0x2] IP6 (flowlabel 0x00001, hlim 64, next-header TCP (6) payload length: 120) ::ffff:10.9.8.7.8 > ::ffff:10.1.2.5.50006: Flags [.], cksum 0x9a12 (correct), seq 1:101, ack 101, win 8192, length 100
	0x0000:  0201 0203 0405 0209 0807 0605 8864 0000  .............d..
	0x0010:  0002 00a2 0057 6000 0001 0078 0640 0000  .....W`....x.@..
	0x0020:  0000 0000 0000 0000 ffff 0a09 0807 0000  ................
	0x0030:  0000 0000 0000 0000 ffff 0a01 0205 0008  ................
	0x0040:  c356 0000 0002 0000 0066 5010 2000 9a12  .V.......fP.....
	0x0050:  0000 4c6f 7265 6d20 6970 7375 6d20 646f  ..Lorem.ipsum.do
	0x0060:  6c6f 7220 7369 7420 616d 6574 2c20 636f  lor.sit.amet,.co
	0x0070:  6e73 6563 7465 7475 7220 6164 6970 6973  nsectetur.adipis
	0x0080:  6369 6e67 2065 6c69 742e 2045 7469 616d  cing.elit..Etiam
	0x0090:  2073 7573 6369 7069 742c 2071 7561 6d20  .suscipit,.quam.
	0x00a0:  6174 2073 6365 6c65 7269 7371 7565 2075  at.scelerisque.u
	0x00b0:  6c74 7269 6369                           ltrici
 [ver 0] [type 0]PPPoE  [ses 0x2] IP6 (flowlabel 0x00001, hlim 64, next-header TCP (6) payload length: 20) ::ffff:10.1.2.5.50006 > ::ffff:10.9.8.7.8: Flags [F.], cksum 0xad93 (correct), seq 101, ack 101, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 8864 0000  .............d..
	0x0010:  0002 003e 0057 6000 0001 0014 0640 0000  ...>.W`......@..
	0x0020:  0000 0000 0000 0000 ffff 0a01 0205 0000  ................
	0x0030:  0000 0000 0000 0000 ffff 0a09 0807 c356  ...............V
	0x0040:  0008 0000 0066 0000 0066 5011 2000 ad93  .....f...fP.....
	0x0050:  0000                                     ..
 [ver 0] [type 0]PPPoE  [ses 0x2] IP6 (flowlabel 0x00001, hlim 64, next-header TCP (6) payload length: 20) ::ffff:10.9.8.7.8 > ::ffff:10.1.2.5.50006: Flags [F.], cksum 0xad92 (correct), seq 101, ack 102, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 8864 0000  .............d..
	0x0010:  0002 003e 0057 6000 0001 0014 0640 0000  ...>.W`......@..
	0x0020:  0000 0000 0000 0000 ffff 0a09 0807 0000  ................
	0x0030:  0000 0000 0000 0000 ffff 0a01 0205 0008  ................
	0x0040:  c356 0000 0066 0000 0067 5011 2000 ad92  .V...f...gP.....
	0x0050:  0000                                     ..
 [ver 0] [type 0]PPPoE  [ses 0x2] IP6 (flowlabel 0x00001, hlim 64, next-header TCP (6) payload length: 20) ::ffff:10.1.2.5.50006 > ::ffff:10.9.8.7.8: Flags [.], cksum 0xad92 (correct), ack 102, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 8864 0000  .............d..
	0x0010:  0002 003e 0057 6000 0001 0014 0640 0000  ...>.W`......@..
	0x0020:  0000 0000 0000 0000 ffff 0a01 0205 0000  ................
	0x0030:  0000 0000 0000 0000 ffff 0a09 0807 c356  ...............V
	0x0040:  0008 0000 0067 0000 0067 5010 2000 ad92  .....g...gP.....
	0x0050:  0000                                     ..
